fsspec/implementations/http.py

```diff
@@ -666,29 +722,38 @@ async def _file_size(url, session=None, size_policy="head", **kwargs):
     head = kwargs.get("headers", {}).copy()
     head["Accept-Encoding"] = "identity"
     kwargs["headers"] = head
-    session = session or await get_client()
+
+    info = {}
     if size_policy == "head":
         r = await session.head(url, allow_redirects=ar, **kwargs)
     elif size_policy == "get":
         r = await session.get(url, allow_redirects=ar, **kwargs)
     else:
         raise TypeError('size_policy must be "head" or "get", got %s' "" % size_policy)
     async with r:
-        try:
-            r.raise_for_status()
+        r.raise_for_status()
+
+        # TODO:
+        #  recognise lack of 'Accept-Ranges',
+        #                 or 'Accept-Ranges': 'none' (not 'bytes')
+        #  to mean streaming only, no random access => return None
+        if "Content-Length" in r.headers:
+            info["size"] = int(r.headers["Content-Length"])
+        elif "Content-Range" in r.headers:
+            info["size"] = int(r.headers["Content-Range"].split("/")[1])
+
+        checksum = r.headers.get("ETag") or r.headers.get("Content-MD5")
```

Comment:
Do you ever have both? 
Since we are now returning a set of information, maybe can consider what other likely keys are of interest (without assuming their format or otherwise processing them).

Reply:
> Do you ever have both?

You can technically get both of them, since these are basically headers.

> Since we are now returning a set of information, maybe can consider what other likely keys are of interest (without assuming their format or otherwise processing them).

This is what we have been using so far for DVC's identification, but can check other keys as well, sure.